### PR TITLE
Android TabbedPageHandler

### PR DESCRIPTION
**Pull request by PureWeen**
### Description of Change ###

This moves TabbedPageRenderer into a Controls Handler. I was hoping to not have to convert TabbedPageRenderer to a handler but the way TabbedPageRenderer does layout is incompatible with our current Navigation Handler.

This PR  moves us closer towards just letting Android naturally layout all the elements. If you compare `TabbedPageRenderer` to `TabbedPageManager` you'll notice I've deleted all the measure/layout code. Now, all the elements are just part of the root `CoordinatorLayout` which allows Android to just natively layout all the elements. This also lights up the features of `AppBarLayout` like Scrolling.

This also updates us to ViewPager2 (ViewPager is no longer in development)

Forms currently doesn't rely on Android to perform any of the page layouts. It collects all the pieces (AppBar, TabLayout) and then calculates where to put them on the screen. This is also a big reason why our Page has a `ContainerArea` property, because the native platform has no way to just calculate this and report it back naturally.

#### Forms view hierarchy
image

#### Maui view hierarchy
image

### What this PR doesn't do
I still need to build out the work for nesting NavigationPages inside TabbedPages but that PR was getting too big so we'll start here.

### PR Checklist ###

- [ ] Targets the correct branch 
- [ ] Tests are passing (or failures are unrelated)
- [ ] Targets a single property for a single control (or intertwined few properties)
- [ ] Adds the property to the appropriate interface
- [ ] Avoids any changes not essential to the handler property
- [ ] Adds the mapping to the PropertyMapper in the handler
- [ ] Adds the mapping method to the Android, iOS, and Standard aspects of the handler
- [ ] Implements the actual property updates (usually in extension methods in the Platform section of Core)
- [ ] Tags ported renderer methods with [PortHandler]
- [ ] Adds an example of the property to the sample project (MainPage)
- [ ] Adds the property to the stub class
- [ ] Implements basic property tests in DeviceTests

#### Does this PR touch anything that might affect accessibility?
- [ ] Does this PR introduce a new control? (If yes, add an example using SemanticProperties to the SemanticsPage)
- [ ] APIs that modify focusability?
- [ ] APIs that modify any text property on a control?
- [ ] Does this PR modify view nesting or view arrangement in anyway?
- [ ] Is there the smallest possibility that your PR will change accessibility? 
- [ ] I'm not sure, please help me

If any of the above checkboxes apply to your PR, then the PR will need to provide testing to demonstrate that accessibility still works.

### CODE REVIEW

File: src/Controls/samples/Controls.Sample/Pages/CustomNavigationPage.xaml.cs [hunk 1]
```diff
@@ -7,29 +7,14 @@ namespace Maui.Controls.Sample.Pages
...
 			InitializeComponent();
 		}
-
-		protected override void OnHandlerChanged()
-		{
-			base.OnHandlerChanged();
-
-#if __ANDROID__
-
-			if (Handler?.NativeView is Android.Views.View layout)
```
Comment (PureWeen) [inline src/Controls/samples/Controls.Sample/Pages/CustomNavigationPage.xaml.cs:21]: Removing this for now. The AppBar behaves oddly if you navigate to a new page and it was hidden on the previous page.

**Comment by kcrg** [comment]
@PureWeen Will these hierarchy changes apply to Shell as well?

**Comment by PureWeen** [comment]
@kcrg Shell already uses these hierarchies

https://github.com/dotnet/maui/blob/b84546026822d63dcad3c9f8315646810a5c1fc6/src/Controls/src/Core/Platform/Android/Resources/layout/shellrootlayout.axml

**Comment by kcrg** [comment]
> @kcrg Shell already uses these hierarchies
> 
> https://github.com/dotnet/maui/blob/b84546026822d63dcad3c9f8315646810a5c1fc6/src/Controls/src/Core/Platform/Android/Resources/layout/shellrootlayout.axml

@PureWeen Is it possible to put android's BottomNavigationView to this CoordinatorLayout? I think I need this to implement hiding on scroll behavior for both AppBarLayout and BottomNavigationView. I tried achieved that but because in XF AppBarLayout and BottomNavigationView are in separate layouts it's impossible - see: https://github.com/xamarin/Xamarin.Forms/issues/13338#issuecomment-883558841

I'm not 100% sure, maybe I miss something.

**Comment by PureWeen** [comment]
@kcrg Interesting, thank you for all the research on that issue! I'll need to poke through those shell layouts a bit and see if there was a motivation to the current setup.

My hope here is to eventually have NavigationPage/TabbedPage/Shell/FlyoutPage just use all the same handler logic. 

I've logged an issue here on the MAUI side
https://github.com/dotnet/maui/issues/2456

File: src/Core/src/Platform/Android/Navigation/NavigationViewFragment.cs [hunk 2]
```diff
@@ -14,10 +15,24 @@ public class NavigationViewFragment : Fragment
 		NavigationLayout? _navigationLayout;
 		FragmentContainerView? _fragmentContainerView;
 
-		NavigationLayout NavigationLayout =>
-			_navigationLayout ??= (FragmentContainerView.Parent as NavigationLayout)
-			?? (FragmentContainerView.Parent?.Parent as NavigationLayout)
-			?? throw new InvalidOperationException($"NavigationLayout cannot be null here");
+		// TODO FIX ME
```
Comment (hartez) [inline src/Core/src/Platform/Android/Navigation/NavigationViewFragment.cs:18]: Did this get pulled over from old code, or is it a new TODO? If it's a new one, can we get some more context/info on what needs to be fixed?

**Review by hartez — CHANGES_REQUESTED** [review]

File: src/Core/src/Platform/Android/Navigation/NavigationViewFragment.cs (hunk 2 above)
Comment (PureWeen) [reply src/Core/src/Platform/Android/Navigation/NavigationViewFragment.cs]: 👍

**Review by hartez — APPROVED** [review]

**Comment by maonaoda** [comment]
@PureWeen 
Sorry~
Why can’t I find the `src/Controls/src/Core/Handlers/TabbedPage/TabbedPageHandler.Android.cs` file at all now?

**Pull request merged**